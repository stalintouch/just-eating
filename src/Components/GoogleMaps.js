/* global google */
import React, { Component } from 'react';

// this component creates a google map with the received props
export default class GoogleMaps extends Component {
  
  componentDidMount() {
    const {lat, lng} = this.props
    var map = new google.maps.Map(this.refs.map, {
      zoom: 12,
      center: {
        lat,
        lng
      }
    });
    var myCenter = new google.maps.LatLng(lat, lng);

    var marker = new google.maps.Marker({
      position: myCenter,
    });
    marker.setMap(map)
  }

  render() {
    return <div ref="map" style={{ height: '100%', width: '100%' }} />;
  }
}
