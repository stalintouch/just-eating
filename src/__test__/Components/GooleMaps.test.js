import GoogleMaps from '../../Components/GoogleMaps';

describe('<GoogleMaps />', () => {
  let wrapperShallow;
  let wrapperMount;

  it('renders correctly', () => {
    wrapperShallow = shallow(<GoogleMaps />);
    expect(wrapperShallow).toMatchSnapshot();
  });
});
