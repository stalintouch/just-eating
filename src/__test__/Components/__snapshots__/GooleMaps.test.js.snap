// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoogleMaps /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleMaps />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "style": Object {
        "height": "100%",
        "width": "100%",
      },
    },
    "ref": "map",
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "style": Object {
          "height": "100%",
          "width": "100%",
        },
      },
      "ref": "map",
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
